internal/util/util.go

```diff
@@ -159,3 +161,27 @@ func GetResourceOwner(owner *string) (string, string, error) {
 
 	return parts[0], parts[1], nil
 }
+
+// CreateJitter creates a random jitter duration in the range [0, max) * duration
+func CreateJitterDuration(max int64, duration time.Duration) (time.Duration, error) {
+	if max <= 0 {
+		return 0, fmt.Errorf("max must be positive")
+	}
+
+	// Generate a random number in the range [0, max)
+	n, err := rand.Int(rand.Reader, big.NewInt(max))
+	if err != nil {
+		return 0, err
+	}
+
+	return (duration * time.Duration(n.Int64())), nil
+}
+
+// NewTickerWithJitter creates a new ticker with a random jitter duration in the range [0, max) * duration
+func NewTickerWithJitter(d time.Duration, jitterDuration time.Duration, maxJitter int64) (*time.Ticker, error) {
```

Comment:
This implementation isn't a jitter anymore, i.e. random fluctuations in the interval between ticks that still have a mean interval of d, but a constant interval between ticks that's been chosen once at startup from [d, d + [0, j_max] * j)). One consequence is higher sync and lower staggering of requests, but more importantly it'd be nice to have d as the mean interval to be able to reason about the load of the system independently of jitter (e.g. a system with d=60s and 100k agents should see a load of 1.7k requests per second).

Reply:
true this only randomizes the start time, we can move back to jitterbug

Reply:
True, I had the same thought but I didn’t consider it of importance, but @fzdarsky you are right, it’s statistically important